Feature/measurement graphing

Andrea-Papaleo opened the pull request.

Andrea-Papaleo committed:
Adds plotting functionality
- generalizes custom tabs

Andrea-Papaleo committed:
updates measurement tree structure
- updates meaaurements when changes made elsewhere
- changes "ListItem button" to "ListItemButton" per updated MUI
- fixes delete category bug in imageviewer

gnodar01 reviewed (changes requested):
Shouldn't Y-axis be X-axis instead?

<img width="900" alt="Screenshot 2024-10-04 at 4 34 23 PM" src="https://github.com/user-attachments/assets/e0304a2b-4667-4df6-b214-bc85a22e1dfb">

Andrea-Papaleo committed:
fixes formatting and label spacing

Andrea-Papaleo committed:
wip adds new Plotcontainer component to display save button

Andrea-Papaleo committed:
renames file

Andrea-Papaleo committed:
[wip] adds plt exporter

gnodar01 committed:
[fix] Save after load

- image must finish loading before attempting to draw it on canvas and
  convert to dataurl

Andrea-Papaleo committed:
adds padding to save button
